src/program.hpp

```diff
@@ -653,6 +664,11 @@ struct cvk_program : public _cl_program, api_object<object_magic::program> {
     std::mutex m_lock;
     std::unique_ptr<std::thread> m_thread;
     std::string m_source;
+#ifndef CLSPV_ONLINE_COMPILER
+    std::vector<uint8_t> m_ir;
+#else
+    std::string m_ir;
+#endif
```

Comment:
I don't think we need two different definitions here. Maybe handling vectors in the file saving utilities would be easier?